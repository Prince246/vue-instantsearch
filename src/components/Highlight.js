import { getPropertyByPath } from '../util/object';
import { warn } from '../util/warn';

export default {
  functional: true,
  props: {
    hit: {
      type: Object,
      required: true,
    },
    attribute: {
      type: String,
      required: true,
    },
  },
  render(h, ctx) {
    const hit = ctx.props.hit;
    const attribute = ctx.props.attribute;

    const attributePath = `_highlightResult.${attribute}.value`;
    const attributeValue = getPropertyByPath(hit, attributePath);

    if (process.env.NODE_ENV !== 'production' && attributeValue === undefined) {
      warn(
        `The "${attribute}" attribute might currently not be configured to be highlighted in Algolia.
        See https://www.algolia.com/doc/api-reference/api-parameters/attributesToHighlight/.`
      );
    }

    return h('span', {
      class: {
        'ais-Highlight': true,
      },
      domProps: {
        innerHTML: attributeValue,
      },
    });
  },
};
