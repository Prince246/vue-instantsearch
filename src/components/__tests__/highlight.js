import { mount } from '@vue/test-utils';
import Highlight from '../Highlight';

afterEach(() => {
  process.env.NODE_ENV = 'test';
});

test('renders proper HTML', () => {
  const result = {
    _highlightResult: {
      attr: {
        value: `con<em>ten</em>t`,
      },
    },
  };

  const wrapper = mount(Highlight, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render an empty string in production if attribute is not highlighted', () => {
  process.env.NODE_ENV = 'production';
  const result = {
    _highlightResult: {},
  };

  const wrapper = mount(Highlight, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should warn when not in production if attribute is not highlighted', () => {
  global.console.warn = jest.fn();

  const result = {
    _highlightResult: {},
  };

  mount(Highlight, {
    context: {
      props: {
        attributeName: 'attr',
        result,
      },
    },
  });

  expect(global.console.warn).toHaveBeenCalledTimes(1);
});

test('allows usage of dot delimited path to access nested attribute', () => {
  const result = {
    _highlightResult: {
      attr: {
        nested: {
          value: `nested <em>val</em>`,
        },
      },
    },
  };

  const wrapper = mount(Highlight, {
    context: {
      props: {
        attributeName: 'attr.nested',
        result,
      },
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
