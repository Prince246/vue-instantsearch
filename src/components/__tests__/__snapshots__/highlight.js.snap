// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows usage of dot delimited path to access nested attribute 1`] = `

<span class="ais-Highlight">
  nested
  <em>
    val
  </em>
</span>

`;

exports[`renders proper HTML 1`] = `

<span class="ais-Highlight">
  con
  <em>
    ten
  </em>
  t
</span>

`;

exports[`should render an empty string in production if attribute is not highlighted 1`] = `

<span class="ais-Highlight">
</span>

`;
